app/utils/profileManagement.js

```diff
@@ -109,6 +124,22 @@ export async function saveProfile(ethereumService, user, profile) {
   ]);
 }
 
+export async function getUserRatings(ethereumService, user, communities) {
+  if (!communities || !user) {
+    return [];
+  }
+  const ratings = [];
+  await Promise.all(
+    communities?.map(async community => {
+      ratings[community.id] = await ethereumService.getUserRating(
```

Comment:
also if you are setting to array via this ratings[community.id]
maybe it's better to use object? or Array.prototype.push?

Reply:
In case of deleted community it could be an error with indexes

Reply:
Array is not for this intentions, you should use object instead, or via array indexes which have strict order